x-pack/plugins/maps/server/index_data.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from 'kibana/server';
+import { Logger } from 'src/core/server';
+import {
+  INDEX_META_DATA_CREATED_BY,
+  ImportResponse,
+  ImportFailure,
+  Settings,
+  Mappings,
+} from '../common';
+import { IndexPatternsService } from '../../../../src/plugins/data/common';
+
+export type InputData = any[];
+interface BodySettings {
+  [key: string]: any;
+}
+
+const DEFAULT_SETTINGS = { number_of_shards: 1 };
+const DEFAULT_MAPPINGS = {
+  _meta: {
+    created_by: INDEX_META_DATA_CREATED_BY,
+  },
+};
+
+export function indexDataProvider(
+  { asCurrentUser }: IScopedClusterClient,
+  indexPatternsService: IndexPatternsService,
+  logger: Logger
+) {
+  async function indexData(
```

Comment:
This method is very generic for creating an index, but the use case is not generic. We are creating a drawing layer so we know the mapping and what is going into the index. The route should not having any pipeline complexities or require mappings. I think the API should be really simple - what is the index name, we know the rest

Reply:
Mappings was one of the few complexities I left in vs. something like GeoJSON Upload. We're still planning on allowing users to do points, shapes and possibly ids, so mappings will change. I agree on pipeline complexities and haven't included these but we should keep mappings for this endpoint

Reply:
We're not handling adding docs yet. Keeping it simple at first. Just index creation

Reply:
but, why not just index everything as geo_shape and have the mappings for ids built in. So the mappings for all drawing layers are the same?

Reply:
We're handling both shapes and points. I guess we could call everything `geo_shape` but my thinking was why not handle both? It's not hard to pass simple mappings from the UI. The nature of IDs and labels is still a subject of discussion so no strong feelings there except that it seems safer to leave that open for now.